server/controllers/gqlController.js

```diff
@@ -108,21 +109,118 @@ gqlController.makeSchemaQueries = async function (req, res, next) {
+                        resolvers[resolversTablePropKey][camelCase(fkObj.foreign_table)] = `(pTable) => {\n      const query = 'SELECT * FROM ${fkObj.foreign_table} WHERE ${fkObj.fk_columns} = $1';\n      const values = [pTable.${pkName}];\n      return db.query(query, values)\n        .then(data => data.rows)\n        .catch(err => new Error(err));\n    },`
+                    } else {
+                    //if the refing table is join
+                        //look for all the other nonjoins that join references
+                        const otherJoinFkObjs = res.locals.data.foreignKeys.filter(el => el.foreign_table === fkObj.foreign_table && el.primary_table !== key);
+                        for (let j = 0; j< otherJoinFkObjs.length; j++) {
+                            resolvers[resolversTablePropKey][camelCase(otherJoinFkObjs[j].primary_table)] = `(pTable) => {\n      const query = 'SELECT * FROM ${otherJoinFkObjs[j].primary_table} LEFT OUTER JOIN ${otherJoinFkObjs[j].foreign_table} ON ${otherJoinFkObjs[j].primary_table}.${nonJoinTables[otherJoinFkObjs[j].primary_table]} = ${otherJoinFkObjs[j].foreign_table}.${otherJoinFkObjs[j].fk_columns} WHERE ${otherJoinFkObjs[j].foreign_table}.${otherJoinFkObjs[j].fk_columns} = $1';\n      const values = [pTable.${pkName}];\n      return db.query(query, values)\n        .then(data => data.rows)\n        .catch(err => new Error(err));\n    },`
+                        }
+                    }
+                }
+                
+                //if this table is referencing another nonjoin's PK
+                if (fkObj.foreign_table === key) {
+                    resolvers[resolversTablePropKey][camelCase(fkObj.primary_table)] =  `(pTable) => {\n      const query = 'SELECT ${fkObj.primary_table}.* FROM ${fkObj.primary_table} LEFT OUTER JOIN ${key} ON ${fkObj.primary_table}.${nonJoinTables[fkObj.primary_table]} = ${key}.${fkObj.fk_columns} WHERE ${key}.${nonJoinTables[key]} = $1';\n      const values = [pTable.${pkName}];\n      return db.query(query, values)\n        .then(data => data.rows)\n        .catch(err => new Error(err));\n    },`
+                }
+            }
+        } 
+        let output = "const resolvers = {";
+        for(let [key,val] of Object.entries(resolvers)){
+            //add top level properties to output
+            output += `\n  ${key}: {`;
+          //add tier2 props to output for current tier1
+        //  for(let [key,val] of Object.entries(x)) {
+        //      output += `\n    ${key}:`
+        //  }  
+            //console.log(typeof val)
+            for(let [innerKey,innerVal] of Object.entries(val)) {
+              output += (`\n    ${innerKey}: `+ innerVal);
+            }
+            output += `\n  },`;
+        }
+        output += "\n}";  
+        res.locals.resolvers = output;
+        return next();
+        // return resolvers;  
+    } catch (err) {
+        console.log('ERR IN MAKERESOLVERS');
+        return next(err);
+    }
+}
```

Comment:
Thoughts on modularizing this function and importing into the controller? Similar to the controllers above?